src/main/resources/Diagram/DiagramEditSheet.xml

```diff
@@ -688,6 +696,28 @@ define('diagramStore', ['jquery', 'xwiki-meta', 'draw.io', 'xwiki-events-bridge'
     return converter;
   };
 
+  // Override for uploading the image as attachment instead of encode it to Base64
+  var oldImportFiles = EditorUi.prototype.importFiles;
+  EditorUi.prototype.importFiles = function(files, x, y, maxSize, fn, resultFn, filterFn, barrierFn, resizeDialog,
+                                             maxBytes, resampleThreshold, ignoreEmbeddedXml) {
+    var newArgs = Array.prototype.slice.call(arguments);
+    newArgs[4] = function(data, mimeType, x, y, w, h) {
+      if (data.substring(0, 5) == 'data:') {
+        let currentFile = files[0];
+        diagramStore.uploadFile(currentFile, currentFile.name).done(function() {
+          let doc = new XWiki.Document();
+          let documentReference = xm.documentReference;
```

Comment:
You don't need this `documentReference`, right?

Reply:
You're right.